deps/tools/common.mk

```diff
@@ -29,6 +31,7 @@ CMAKE_COMMON += -DCMAKE_CXX_COMPILER="$(CXX_BASE)"
 ifneq ($(strip $(CMAKE_CXX_ARG)),)
 CMAKE_COMMON += -DCMAKE_CXX_COMPILER_ARG1="$(CMAKE_CXX_ARG)"
 endif
+CMAKE_COMMON += -DCMAKE_LINKER="$$(which $(LD))"
```

Comment:
I even left a comment `# The call to which here is to work around https://cmake.org/Bug/view.php?id=14366` - that was fixed in cmake 3.0, we require 3.4.3 these days unless you're using system llvm